bokehjs/src/coffee/core/selection_manager.coffee

```diff
@@ -17,83 +16,32 @@ export class SelectionManager extends HasProps
     @inspectors = {}
     @last_inspection_was_empty = {}
 
-  select: (tool, renderer_views, geometry, final, append=false) ->
-    source = @source
-    if source != renderer_views[0].model.data_source
-      logger.warn('select called with mis-matched data sources')
-
-    #view = renderer_view.model.view
-    #if source != view.source
-    #  logger.warn('select called with view and data source mismatch')
-
-    indices_renderers = []
+  select: (renderer_views, geometry, final, append=false) ->
+    did_hit = false
 
     for r in renderer_views
-      hit_test_result = r.hit_test(geometry)
-      # renderers that don't support hit testing a particular kind of geometry
-      # will return null, need to filter those out and see if anything is left
-      if hit_test_result != null
-        indices_renderers.push(r.model.view.convert_selection_from_subset(hit_test_result))
-
-    if indices_renderers.length == 0
-      return false
-
-    if indices_renderers?
-      indices = indices_renderers[0]
-      for indices_other in indices_renderers
-        indices.update_through_union(indices_other)
-
-      @selector.update(indices, final, append)
-
-      @source.selected = @selector.indices
-
-      source.select.emit()
-
-      return not indices.is_empty()
-    else
-      return false
-
-  inspect: (tool, renderer_view, geometry, data) ->
-    source = @source
-    if source != renderer_view.model.data_source
-      logger.warn('inspect called with mis-matched data sources')
-
-    view = renderer_view.model.view
-    if source != view.source
-      logger.warn('inspect called with view and data source mismatch')
+      did_hit ||= r.hit_test(geometry, final, append)
 
-    indices = renderer_view.model.view.convert_selection_from_subset(renderer_view.hit_test(geometry))
+    @source.select.emit()
 
-    if indices?
-      r_id = renderer_view.model.id
+    return did_hit
 
-      if indices.is_empty()
-        if not @last_inspection_was_empty[r_id]?
-          @last_inspection_was_empty[r_id] = false
-        if @last_inspection_was_empty[r_id]
-          return
-        else
-          @last_inspection_was_empty[r_id] = true
-      else
-        @last_inspection_was_empty[r_id] = false
+  inspect: (renderer_view, geometry) ->
+    did_hit = false
+    @_set_inspector(renderer_view)
 
-      inspector = @_get_inspector(renderer_view)
-      inspector.update(indices, true, false, true)
+    did_hit ||= renderer_view.hit_test(geometry, false, false, "inspect")
 
-      @source.setv({inspected: inspector.indices}, {"silent": true })
+    @last_inspection_was_empty[renderer_view.model.id] = did_hit
```

Comment:
It's not clear that this property is used for anything. It might be better to remove.